CDTDatastore/query/CDTQProjectedDocumentRevision.h

```diff
@@ -28,12 +30,14 @@
 /**
  Initialise with a datastore so mutableCopy can return a full document.
  */
-- (id)initWithDocId:(NSString *)docId
-         revisionId:(NSString *)revId
-               body:(NSDictionary *)body
-            deleted:(BOOL)deleted
-        attachments:(NSDictionary *)attachments
-           sequence:(SequenceNumber)sequence
-          datastore:(CDTDatastore *)datastore;
+- (nullable instancetype)initWithDocId:(NSString *)docId
```

Comment:
It would result in more `if let` statements although this being said, I've taken a quick look at the header of NSString, and it returns a nonnull value for init. Here's an extract that shows it:

```objc
@interface NSString : NSObject <NSCopying, NSMutableCopying, NSSecureCoding>

#pragma mark *** String funnel methods ***

/* NSString primitives. A minimal subclass of NSString just needs to implement these two, along with an init method appropriate for that subclass. We also recommend overriding getCharacters:range: for performance.
 */
@property (readonly) NSUInteger length;
- (unichar)characterAtIndex:(NSUInteger)index;

/* The initializers available to subclasses. See further below for additional init methods.
*/
- (instancetype)init NS_DESIGNATED_INITIALIZER;
- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
```

Its in an audited section, so I think if its good enough for Apple it should probably be good enough for us. Any thoughts @brynh?